dlt/destinations/impl/sqlalchemy/db_api_client.py

```diff
@@ -0,0 +1,378 @@
+            if args:
+                # Sqlalchemy text supports :named paramstyle for all dialects
+                query, kwargs = self._to_named_paramstyle(query, args)  # type: ignore[assignment]
+                args = ()
+            query = sa.text(query)
+        with self._transaction():
+            yield SqlaDbApiCursor(self._current_connection.execute(query, *args, **kwargs))  # type: ignore[abstract]
+
+    def get_existing_table(self, table_name: str) -> Optional[sa.Table]:
+        """Get a table object from metadata if it exists"""
+        key = self.dataset_name + "." + table_name
+        return self.metadata.tables.get(key)  # type: ignore[no-any-return]
+
+    def create_table(self, table_obj: sa.Table) -> None:
+        table_obj.create(self._current_connection)
+
+    def _make_qualified_table_name(self, table: sa.Table, escape: bool = True) -> str:
+        if escape:
+            return self.dialect.identifier_preparer.format_table(table)  # type: ignore[attr-defined,no-any-return]
+        return table.fullname  # type: ignore[no-any-return]
+
+    def make_qualified_table_name(self, table_name: str, escape: bool = True) -> str:
+        tbl = self.get_existing_table(table_name)
+        if tbl is None:
+            tmp_metadata = sa.MetaData()
+            tbl = sa.Table(table_name, tmp_metadata, schema=self.dataset_name)
+        return self._make_qualified_table_name(tbl, escape)
+
+    def alter_table_add_column(self, column: sa.Column) -> None:
+        """Execute an ALTER TABLE ... ADD COLUMN ... statement for the given column.
+        The column must be fully defined and attached to a table.
+        """
+        # TODO: May need capability to override ALTER TABLE statement for different dialects
+        alter_tmpl = "ALTER TABLE {table} ADD COLUMN {column};"
+        statement = alter_tmpl.format(
+            table=self._make_qualified_table_name(self._make_qualified_table_name(column.table)),
+            column=self.compile_column_def(column),
+        )
+        self.execute_sql(statement)
+
+    def escape_column_name(self, column_name: str, escape: bool = True) -> str:
+        if self.dialect.requires_name_normalize:
+            column_name = self.dialect.normalize_name(column_name)
+        if escape:
+            return self.dialect.identifier_preparer.format_column(sa.Column(column_name))  # type: ignore[attr-defined,no-any-return]
+        return column_name
+
+    def compile_column_def(self, column: sa.Column) -> str:
+        """Compile a column definition including type for ADD COLUMN clause"""
+        return str(sa.schema.CreateColumn(column).compile(self.engine))
+
+    def reflect_table(
+        self,
+        table_name: str,
+        metadata: Optional[sa.MetaData] = None,
+        include_columns: Optional[Sequence[str]] = None,
+    ) -> Optional[sa.Table]:
+        """Reflect a table from the database and return the Table object"""
+        if metadata is None:
```

Comment:
there's no way to reflect many tables at once? still - this is good enough :)

Reply:
There is `metadata.reflect(only=[...table_names])`, but the problem with that is the whole operation fails if any of the tables don't exist. So it's tricky (unless we start using un-documented internals in sqla to hack something together).   
 But I *think* internally it handles one table at a time anyway so it shouldn't be much difference.